src/ValueParsers/DecimalParser.php

```diff
@@ -66,9 +65,9 @@ private function getMath() {
 	 *         does not use scientific notation, $exponent will be 0.
 	 */
 	public function splitDecimalExponent( $valueString ) {
-		if ( preg_match( '/^(.*)([eE]|x10\^)([-+]?[,\d]+)$/', $valueString, $matches ) ) {
-			$exponent = $this->normalizeDecimal( $matches[3] );
-			return array( $matches[1], intval( $exponent ) );
+		if ( preg_match( '/^(.*)(?:[eE]|x10\^)([-+]?[\d,]+)$/', $valueString, $matches ) ) {
```

Comment:
I wonder why we allow commas here... there aren't any tests for this method, so I have no idea what kind of input would that be. Also Wikibase's MediaWikiNumberUnlocalizer::getNumberRegex doesn't match commas in the exponent, while it says "Keep in sync with DecimalParser::splitDecimalExponent.". Unrelated to this change, but I wanted to let you know.